internal/armada/scheduling/lease.go

```diff
@@ -325,6 +363,31 @@ func (c *leaseContext) leaseJobs(
 	return jobs, slice, nil
 }
 
+func isJobSchedulable(c *leaseContext, job *api.Job, remainder common.ComputeResourcesFloat, candidatesLimit LeasePayloadLimit) bool {
+	isJobLargeEnough := isLargeEnough(job, c.minimumJobSize)
+	isRemainderValid := remainder.IsValid()
+	isCandidateWithinLimit := candidatesLimit.IsWithinLimit(job)
+
+	isRegularlySchedulable := isJobLargeEnough && isRemainderValid && isCandidateWithinLimit
+	isPreemptiveJob := isJobLargeEnough && hasPriorityClass(job.PodSpec)
+
+	return isRegularlySchedulable || isPreemptiveJob
+}
+
+// validateOrDefaultPriorityClass checks is the pod spec's priority class configured as supported in Server config
+// if not, default to DefaultPriorityClass if it is specified
+// otherwise default to no Priority Class
+func validateOrDefaultPriorityClass(podSpec *v1.PodSpec, preemptionConfig configuration.PreemptionConfig) {
+	_, ok := preemptionConfig.PriorityClasses[podSpec.PriorityClassName]
```

Comment:
```suggestion
       if preemptionConfig.PriorityClasses == nil {
                   podSpec.PriorityClassName = preemptionConfig.DefaultPriorityClass
        }
	_, ok := preemptionConfig.PriorityClasses[podSpec.PriorityClassName]
```

Reply:
If no priority classes are defined, I guess the map would be nil. If that's the case, this will protect against panics.

Reply:
If you agree with the change, you may need to fix the spacing here. I'm in a browser...